src/coda/apps/breadcrumbs/decorators.py

```diff
@@ -0,0 +1,178 @@
+from functools import wraps
+from typing import Any, Callable
+from urllib.parse import parse_qs, urlparse
+
+from django.http import HttpRequest, HttpResponse
+from django.urls import reverse
+from django.utils.http import urlencode as django_urlencode
+
+# Type for title - can be string or callable that takes request, *args, **kwargs
+TitleType = str | Callable[[HttpRequest, Any, Any], str]
+
+
+def breadcrumb(
```

Comment:
I have only have one thing, I think I would like to change. Currently you are using breadcrumb for view functions and `@method_decorator(breadcrumb(...))` for classes. I think it would be cool if the breadcrumb decorator was smart enough to figure out if we have a function or class on its own.

So in the end, the decorator shouldn't care whether we have a function or class and just be able to apply `@method_decorator` on its own if it receives a class